python/tank/authentication/login_dialog.py

```diff
@@ -219,27 +285,47 @@ def _ok_pressed(self):
         if len(site) == 0:
             self._set_error_message(self.ui.message, "Please enter the address of the site to connect to.")
             return
-        if len(login) == 0:
-            self._set_error_message(self.ui.message, "Please enter your login name.")
-            return
-        if len(password) == 0:
-            self._set_error_message(self.ui.message, "Please enter your password.")
-            return
+        if not self._use_sso:
+            if len(login) == 0:
+                self._set_error_message(self.ui.message, "Please enter your login name.")
+                return
+            if len(password) == 0:
+                self._set_error_message(self.ui.message, "Please enter your password.")
+                return
 
         # if not protocol specified assume https
         if len(site.split("://")) == 1:
             site = "https://%s" % site
             self.ui.site.setText(site)
 
+        session_token = None
+        if self._use_sso:
+            login = ""
+            password = ""
+            res = self._saml2_sso.on_sso_login_attempt({
+                "host": site,
+                "cookies": self._cookies,
+                "product": "toolkit"
+            })
+            if res == QtGui.QDialog.Accepted:
+                session_token = self._saml2_sso._session.session_id
```

Comment:
I got really confused a couple of lines below when I saw session_token being passed down when authenticating.